x/incentive/simulation/genesis.go

```diff
@@ -0,0 +1,150 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/kava-labs/kava/x/incentive/types"
+)
+
+var (
+	CollateralDenoms    = [3]string{"bnb", "xrp", "btc"}
+	RewardDenom         = "ukava"
+	MaxTotalAssetReward = sdk.NewInt(1000)
+)
+
+// RandomizedGenState generates a random GenesisState for incentive module
+func RandomizedGenState(simState *module.SimulationState) {
+	params := GenParams(simState.Rand)
+	rewardPeriods := GenRewardPeriods(simState.Rand, simState.GenTimestamp, params.Rewards)
+	claimPeriods := GenClaimPeriods(rewardPeriods)
+	claimPeriodIDs := GenNextClaimPeriodIds(claimPeriods)
+
+	// New genesis state holds valid, linked reward periods, claim periods, and claim period IDs
+	incentiveGenesis := types.NewGenesisState(params, types.DefaultPreviousBlockTime,
+		rewardPeriods, claimPeriods, types.Claims{}, claimPeriodIDs)
+	if err := incentiveGenesis.Validate(); err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Selected randomly generated %s parameters:\n%s\n", types.ModuleName, codec.MustMarshalJSONIndent(simState.Cdc, incentiveGenesis))
+	simState.GenState[types.ModuleName] = simState.Cdc.MustMarshalJSON(incentiveGenesis)
+}
+
+// GenParams generates random rewards and is active by default
+func GenParams(r *rand.Rand) types.Params {
+	params := types.NewParams(true, GenRewards(r))
+	if err := params.Validate(); err != nil {
+		panic(err)
+	}
+	return params
+}
+
+// GenRewards generates rewards for each specified collateral type
+func GenRewards(r *rand.Rand) types.Rewards {
+	var rewards types.Rewards
+	for _, denom := range CollateralDenoms {
+		active := true
+		// total reward is in range (half max total reward, max total reward)
+		amount := simulation.RandIntBetween(r, int(MaxTotalAssetReward.Int64()/2), int(MaxTotalAssetReward.Int64()))
+		totalRewards := sdk.NewInt64Coin(RewardDenom, int64(amount))
+		// generate a random number of days between 1-2 to use for reward's times
+		numbDays := simulation.RandIntBetween(r, 1, 2) * 24
+		duration := time.Duration(time.Hour * time.Duration(numbDays))
+		timeLock := time.Duration(time.Hour * time.Duration(numbDays/2)) // half as long as duration
+		claimDuration := time.Hour * time.Duration(numbDays*2)           // twice as long as duration
```

Comment:
```suggestion
		numbHours := simulation.RandIntBetween(r, 6, 48)
		duration := time.Duration(time.Hour * time.Duration(numbHours))
		timeLock := time.Duration(time.Hour * time.Duration(numbHours/2)) // half as long as duration
		claimDuration := time.Hour * time.Duration(numbHours*2)           // twice as long as duration
```

Suggestion: shorten the minimum period length so there can be more reward claims per sim.

Reply:
https://github.com/Kava-Labs/kava/pull/439/commits/b47bb2f59a9bd19d17a0f6e6e0d4946dfb4b79aa